celavi/pylca_celavi/pylca_opt_background.py

```diff
@@ -308,92 +61,37 @@ def solver_optimization(tech_matrix,F):
-            df = pd.DataFrame.from_dict(model.s.extract_values(), orient='index', columns=[str(model.s)])
-            return df
-          
-        # This is an optional code path that allows the script to be run outside of
-        # pyomo command-line.  For example:  python transport.py
-    
-        # This emulates what the pyomo command-line tools does
-        opt = SolverFactory("glpk")
-        
-        results = opt.solve(model)        
-        solution = pyomo_postprocess(None, model, results)   
-        scaling_vector = pd.DataFrame()
-        #The process emissions do not have any location. That is the reason we have to remove the location information
-        #and then merge with the processes. 
-        scaling_vector['process'] = uslci_process
-        scaling_vector['scaling_factor'] = solution['s']         
-       
         #Calculations for total emissions
-        emissions_results_df = process_emissions.merge(scaling_vector, on = ['process'], how = 'left')
-        emissions_results_df['value'] = emissions_results_df['value'] * emissions_results_df['scaling_factor']
-        emissions_results_df = emissions_results_df[emissions_results_df['value'] > 0]
-        emissions_results_df = emissions_results_df.fillna(0)
-        #All emission are grouped and summed up together in this dataframe
-        emissions_results_total = emissions_results_df.groupby(by = ['product','unit'])['value'].agg(sum).reset_index()
-    
-        #Calculations for total product flows only
-        #For this calculation we need the locations names as processes are tied to locations
-        scaling_vector['process'] = uslci_process
-        product_results_df = process_product.merge(scaling_vector, on = ['process'], how = 'left')
-        product_results_df['value'] = product_results_df['value'] * product_results_df['scaling_factor']
-        product_results_df = product_results_df[product_results_df['value'] > 0]
-        product_results_df = product_results_df.fillna(0)
-        #All emission are grouped and summed up together in this dataframe
-        product_results_total = product_results_df.groupby(by = ['conjoined_flownames','unit'])['value'].agg(sum).reset_index()
-    
-        return emissions_results_total,product_results_total
-            
+        final_result = scv_df.merge(process_emissions, on = ['process'])
+        final_result['total_emission'] = final_result['scaling vector'] * final_result['value']
+        final_result['value'] = final_result['total_emission']
+        em_result = final_result[['product','value','process','unit']]
+        co2 = em_result[em_result['product'] == 'Carbon dioxide']
+        emissions_results_total = final_result.groupby(['product','unit'])['value'].agg('sum').reset_index()
+      
+        return emissions_results_total
+
     
-    def runner(tech_matrix, F,i,l,j,k,route_id,final_demand_scaler):
+    def runner(tech_matrix, F,i,l,j,k,route_id,final_demand_scaler,process_emissions):
```

Comment:
Minor quibble, doesn't really need to be fixed - There's three different functions in pyLCIA called "runner". Can they be re-named for greater clarity? Can there be just one runner function used throughout pyLCIA?

Reply:
Rectified with new names.